runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -202,36 +230,50 @@ void registerTable(@Nonnull String namespace,
         while (numRetries-- > 0) {
             // Schema validation to ensure that there is either proper modification of the schema across open calls.
             // Or no modification to the protobuf files.
+            /**
+             * Caller is opening a new table with a map of protobufFilename -> protobufFileDescriptor pairs
+             * Here are the cases that can happen:
+             * 1. No entry in TableRegistry -> New table
+             *  1.1: When inserting new protobufs into descriptor table, we find existing entries and they all match
+             *  1.2: When inserting new protobufs into descriptor table, we find mismatching [Schema Change!]
+             *
+             * 2. Entry exists in TableRegistry -> Existing table re-opened:
+             *    2.1: Existing protobuf file list matches, newly inserted protobuf list
+             *    2.1.1: Each of Existing protobuf file descriptors match that in new file descriptor list
+             *    2.1.2: Existing protobuf file descriptor does not match that in new file descriptor![ SchemaChange!]
+             *
+             *    2.2: Existing protobuf file list does not match, newly opened protobuf list [big Schema Change]
+             *    2.2: Existing protobuf file list matches existing proto files
+             *    2.3: Existing protobuf file's descriptor mismatches with the incoming descriptor
+             */
```

Comment:
We need an explicit unit test that targets this logic.

Reply:
Schema update is tested in CorfuStoreShimTest.java
Don't know of an easy way to produce different protobuf files with the same filename, but different schemas.

Reply:
Done added unit tests